lib/reservation_manager.rb

```diff
@@ -0,0 +1,74 @@
+require 'date'
+
+module Hotel
+  class ReservationManager
+    attr_reader :reservations, :block_reservations
+
+    def initialize
+      @reservations = []
+      @block_reservations = []
+    end
```

Comment:
I suggest also keeping `rooms` as an array of numbers and setting it with your `hotel_rooms` method, so you don't have to repeatably call that method.